common/src/main/java/com/microsoft/identity/common/internal/request/AcquireTokenSilentOperationParameters.java

```diff
@@ -61,8 +63,32 @@ public void validate() throws ArgumentException {
     }
 
     private boolean authorityMatchesAccountEnvironment() {
-        final AzureActiveDirectoryCloud cloud = AzureActiveDirectory.getAzureActiveDirectoryCloud(mAccount.getEnvironment());
-        return cloud.getPreferredNetworkHostName().equals(getAuthority().getAuthorityURL().getAuthority());
+        final String methodName = ":authorityMatchesAccountEnvironment";
+        try {
+            if (!AzureActiveDirectory.isInitialized()) {
+                performCloudDiscovery();
+            }
+            final AzureActiveDirectoryCloud cloud = AzureActiveDirectory.getAzureActiveDirectoryCloud(mAccount.getEnvironment());
+            return cloud != null && cloud.getPreferredNetworkHostName().equals(getAuthority().getAuthorityURL().getAuthority());
+        } catch (IOException e) {
+            Logger.error(
+                    TAG + methodName,
+                    "Unable to perform cloud discovery",
+                    e);
+            return false;
+        }
+    }
+
+    private static void performCloudDiscovery() throws IOException {
+        final Object sLock = new Object();
```

Comment:
I don't think this offering any protection as an instance variable....

Reply:
each call would construct a new instance of the lock so it would always be free

Reply:
Addressed